app/src/main/java/com/celzero/bravedns/ui/HomeScreenActivity.kt

```diff
@@ -243,14 +236,23 @@ class HomeScreenActivity : AppCompatActivity() {
         if(day == Calendar.FRIDAY || day == Calendar.SATURDAY) {
             if (numOfDays > 0) {
                 Log.i(LOG_TAG, "App update check initiated, number of days: $numOfDays")
-                checkForAppUpdate(false)
+                checkForUpdate()
                 checkForBlockListUpdate()
             } else {
                 Log.i(LOG_TAG, "App update check not initiated")
             }
         }
     }
 
+    fun checkForUpdate(userInitiation:Boolean = false) {
+        if (persistentState.downloadSource == Constants.DOWNLOAD_SOURCE_PLAY_STORE) {
```

Comment:
Should this stay? With this PR the G-Play update check is only done for apps installed from G-Play, other builds (F-Droid in the future, from website) won't even contain the dependency for it and always use the web updater.
Currently I left this check in here for users who use the G-Play version but did not download from G-Play.

Reply:
We don't need this shared pref value or check anymore.

Reply:
@Ch4t4r Hussain says, this check isn't required (we won't distribute PlayStore flavoured builds from anywhere else). But... PlayStore mirrors like ApkMirror and others might distribute that flavour... In which case, we would need to fallback on the web-updater, right, and not the PlayStore update checks? Or... How is this ideally supposed to work? 

Today, RethinkDNS distributed from the website and PlayStore are signed with the same keys.

Reply:
I had the mirrors in mind as well. You can't know whether the play services are present on those devices in advance -- you could test at runtime though and fallback to the web updater if they aren't.

Reply:
You're right. I guess we need to check for presence of Play Services too (which, if absent, would skip update-checks and show an error toast on Play flavoured builds): https://github.com/googlesamples/google-services/blob/7cbe0486697a/android/gcm/app/src/main/java/gcm/play/android/samples/com/gcmquickstart/MainActivity.java#L103-L117